### ISSUE TITLE
Keyboard Copy (or Cut) 'n' Paste fails in `Data Type Manager` and all `Find Data Types by ...` panels

### ISSUE THREAD

**Issue by Wall-AF** [NationalSecurityAgency/ghidra#6137]
**Describe the bug**
See title.

**To Reproduce**
1. Select a datatype in either the `Data Type Manager` and/or `Find Data Types by ...` panels
2. `Copy` (or `Cut`) it (Ctrl+C or Ctrl+X)
3. Try to `Paste` it elsewhere
4. Nothing happens!

**Expected behavior**
The selected item should be copied (or moved) to the new location!

**Screenshots**
N/A

**Attachments**
N/A

**Environment (please complete the following information):**
 - OS: Windows 11
 - Java Version: 17.0.3.1
 - Ghidra Version: 11
 - Ghidra Origin: official GitHub distro

**Additional context**
FYI, the `Paste` option in the context menu isn't enabled.

**Comment by mumbel** [comment]
`Try to Paste it elsewhere` define "elsewhere".  I've never had issues when you have a category/directory highlighted

**Comment by Wall-AF** [comment]
Sorry, "elsewhere" meaning within the same panel.  I.e. I tried to move a datatype from the root category to inside a subcategory in the same or another .gtd file but it wouldn't. .

**Comment by dragonmacher** [comment]
This seems to be working for me.   Notably, if I try to paste on a non-program archive that is not 'open for editing', then it does not work.   Once I open the archive for editing, it works as expected.

**Comment by Wall-AF** [comment]
Just figured out, it isn't working when there's a filter in use!  I think that shouldn't matter.

**Comment by Wall-AF** [comment]
I'm always usiing a filter otherwise there's too much to scroll through!

**Comment by dragonmacher** [comment]
I too think it should work while filtered.

**Comment by dragonmacher** [comment]
I just tried this in both the `Data Type Manager` and in the `Find Data Types by Name...`.  I had a filter applied in both cases.  It seems to be working as expected.

I copied from the Program archive or from another archive and then pasted into a folder inside of the Program archive.

**Comment by Wall-AF** [comment]
I just took the 2 screenshots below showing visually what's happening.  The one on the left is filtered with `enumD` (using a Regular Expression) and after pressing `Ctrl+X` (cut) the icons remain unchanged and moving the insertion point to another category and pressing `Ctrl+V` (paste) doesn't move those highlighted items.  In the one on the right, I've removed the filter and the _cut_ is clearly indicated by a change in those icons and the _paste_ works.
![image](https://github.com/NationalSecurityAgency/ghidra/assets/67698927/ab412b1f-8f7d-4640-aee8-8d1564711833)
